Project-Aurora/Project-Aurora/Devices/NZXT/NZXTDevice.cs

```diff
@@ -153,8 +157,21 @@ public bool UpdateDevice(Dictionary<DeviceKeys, Color> keycolors, DoWorkEventArg
                     }
                 }
 
-                DeviceLoader.HuePlus?.ApplyEffect(DeviceLoader.HuePlus.Both, new NZXTSharp.Fixed(huepluscolors.ToArray()));
-                DeviceLoader.KrakenX?.ApplyEffect(DeviceLoader.KrakenX.Ring, new NZXTSharp.Fixed(krakenringcolors.ToArray()));
+                if (!huepluscolors.Equals(hue_device) || forced)
```

Comment:
This doesn't check if the two lists are equal, it checks if they are referencing the same object. Since you are creating `huepluscolors` at the start of this method, the references will never be the same. To check if the lists are equal a different method such as `Enumerable.SequenceEqual` should be used.

I've created a [.NET Fiddle](https://dotnetfiddle.net/a6ohls) illustrating this in this context.